src/main/java/htsjdk/tribble/readers/BlockCompressedAsciiLineReader.java

```diff
@@ -0,0 +1,42 @@
+package htsjdk.tribble.readers;
+
+import htsjdk.samtools.util.BlockCompressedInputStream;
+import htsjdk.samtools.util.CloserUtil;
+
+import java.io.IOException;
+
+/**
+ * An AsciiLineReader implementation that wraps a BlockCompressedInputStream and provides no additional buffering.
+ * Useful for cases where we need to preserve virtual file pointers in the underlying stream, such as during indexing.
+ */
+public class BlockCompressedAsciiLineReader extends AsciiLineReader {
```

Comment:
If the static method is done in `AsciiLineReader`, this could be a private inner class to avoid direct usages.